src/restLink.ts

```diff
@@ -26,9 +30,15 @@ export namespace RestLink {
 
   export type HeadersMergePolicy = (...headerGroups: Headers[]) => Headers;
 
-  export interface FieldNameNormalizer {
+  export interface SimpleFieldNameNormalizer {
     (fieldName: string): string;
   }
+  export interface KeyedFieldNameNormalizer {
+    (fieldName: string, keypath: string[]): string;
+  }
+  export type FieldNameNormalizer =
+    | SimpleFieldNameNormalizer
+    | KeyedFieldNameNormalizer;
```

Comment:
What do you think about consolidating these into a single type? It looks like they have the same signature but accepts an optional `keypath` parameter?

```ts
export interface FieldNameNormalizer {
    (fieldName: string, keypath?: string[]): string;
}
```

Reply:
That’s a great improvement. I’m still new to typescript, and I think I briefly had a syntax error that confused me, leading me to splitting that type into two subtypes. I’ll try to re-combine them, and simplify my PR

Reply:
Implemented!